tests/test_link.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+import properties
+import traitlets
+
+class TestLink(unittest.TestCase):
+
+    def _test_link(self, class_a, class_b):
+
+        hp1 = class_a()
+        hp2 = class_b()
+
+        my_dlink = properties.directional_link((hp1, 'a'), (hp2, 'a'))
+        hp1.a = 5
+        assert hp2.a == 5
+        hp2.a = 100
+        assert hp1.a == 5
+
+        my_dlink.unlink()
+
+        hp1.a = 5
+        assert hp2.a == 100
+        hp2.a = 10
+        assert hp1.a == 5
+
+        my_dlink = properties.directional_link((hp1, 'a'), (hp2, 'a'), update_now=True)
+        assert hp2.a == 5
+        my_dlink.unlink()
+
+        my_dlink = properties.directional_link((hp1, 'a'), (hp2, 'a'), update_now=True, transform=lambda x: 2*x)
+        assert hp2.a == 10
+
+        my_dlink.unlink()
+        hp2.a = 10
+
+        my_link = properties.link((hp1, 'a'), (hp2, 'a'))
+        hp1.a = 6
+        assert hp2.a == 6
+        hp2.a = 100
+        assert hp1.a == 100
+
+        my_link.unlink()
+
+        hp1.a = 5
+        assert hp2.a == 100
+        hp2.a = 10
+        assert hp1.a == 5
+
+        hp3 = class_b(a=0)
+        hp4 = class_a(a=100)
+
+        my_link = properties.link((hp1, 'a'), (hp2, 'a'), (hp3, 'a'), (hp4, 'a'))
+        hp4.a = 100
+        assert hp1.a == 5
+        assert hp2.a == 10
+        assert hp3.a == 0
+
+        hp4.a = 10
+        assert hp1.a == 10
+        assert hp2.a == 10
+        assert hp3.a == 10
+
+        my_link.unlink()
+
+        hp1.a = 5
+
+        my_link = properties.link((hp1, 'a'), (hp2, 'a'), (hp3, 'a'), update_now=True)
+        assert hp2.a == 5
+        assert hp3.a == 5
+
+        my_link.unlink()
+
+        hp3.a = 10
+        my_link = properties.link((hp1, 'a'), (hp2, 'a'), (hp3, 'a'), update_now=True)
+        assert hp3.a == 5
+
+        my_link.unlink()
+
+        hp1.a = 1
+        hp2.a = 2
+        hp3.a = 3
+
+        my_link.relink()
+        assert hp1.a == 1
+        assert hp2.a == 2
+        assert hp3.a == 3
+        hp3.a = 100
+        assert hp2.a == 100
+        assert hp1.a == 100
+        my_link.unlink()
+
+
+    def test_link(self):
+
+        class HP(properties.HasProperties):
+            a = properties.Integer('a')
+
+        class HT(traitlets.HasTraits):
```

Comment:
Yes, it should - the internal logic of `directional_link` is almost identical to traitlets (see https://github.com/ipython/traitlets/blob/4.3.2/traitlets/traitlets.py#L305 ). Tests which only use `HasTraits` are run here to confirm that it works: https://github.com/3ptscience/properties/pull/155/files#diff-8eee08c4b7dd877dbd2f88b5b2f88b99R109